clang/runtime/dpct-rt/include/dpct/math.hpp

```diff
@@ -2216,6 +2218,143 @@ void ldmatrix(uintptr_t addr, T *m1, T *m2, T *m3, T *m4, bool trans = false) {
   ldmatrix(addr, m4, trans, 3);
 }
 
+/// Multiplies 2 matrices (A & B) and adds the result to C matrix and
+/// accumulates the result to a D matrix (MAD). Requires the sub-group size of
```

Comment:
The functionality description for this helper function is not accurate; this helper function only calculates the four elements of the result matrix D for shape and type`m16n8k16 (f32.f16.f16.f32 & s32.s8.s8.s32)`

Reply:
Added more description to the algo functionality